dot/state/storage_notify_test.go

```diff
@@ -120,14 +119,18 @@ func TestStorageState_RegisterStorageChangeChannel_Multi_Filter(t *testing.T) {
 	key1 := []byte("key1")
 	value1 := []byte("value1")
 
+	ts.Set(key1, value1)
+	err = ss.StoreTrie(ts)
+	require.NoError(t, err)
+
 	num := 5
 	chs := make([]chan *SubscriptionResult, num)
 	ids := make([]byte, num)
-	subFilter := make(map[string]bool)
-	subFilter[common.BytesToHex(key1)] = true
 
 	for i := 0; i < num; i++ {
 		chs[i] = make(chan *SubscriptionResult)
+		subFilter := make(map[string][]byte)
+		subFilter[common.BytesToHex(key1)] = []byte{}
```

Comment:
No, moving this to outside the loop causes the test to fail.  Each StorageSubscription holds it's own filter map, which holds cached values for the key.  When a value is updated, the filter map get updated to that value, if a second subscription uses the shared map, the second subscription will not notify because it's value won't be new.

Reply:
You can write the code like shown below as well.
This couples your map with the struct. No need to declare an additional variable.
```
sub := StorageSubscription{
	Filter:   map[string][]byte{
		common.BytesToHex(key1): {},
	},
	Listener: chs[i],
}
```